### PR TITLE
Avoid relying on Astropy version to import old stuff

### LINKED ISSUE #31: [astropy] CI fails on fork due to astropy version check here
Workflow:

* Push changes to `astropy` fork.
* Wait for GitHub Actions to pass on fork without relying on upstream.

What happens:

* Fork has no tags.
* `setuptools_scm` unable to build correct `astropy` version on fork (it is usually like 0.1.devNNNNN).
* This check below triggers outdated import on the fork CI run.

https://github.com/astropy/pytest-astropy-header/blob/53b214e433c38aee6ce43024d2d1d9540d4226ed/pytest_astropy_header/display.py#L42-L56

* This failure appears. Example log: https://github.com/pllim/astropy/runs/1408659448?check_suite_focus=true (for astropy/astropy#11052)

```
  File ".../pytest_astropy_header/display.py", line 47, in <module>
    from astropy.tests.pytest_plugins import (PYTEST_HEADER_MODULES,
ModuleNotFoundError: No module named 'astropy.tests.pytest_plugins'
```

Why we need this to work: @saimn has proposed that maybe we set up GitHub Actions to run exclusively on author's fork for `astropy` PRs instead of having them run both upstream and in the fork.

Possible solution: Drop support for `astropy<4` checks in this plugin.

### PR DESCRIPTION
**Pull request by saimn**
Fix #31.

The drawback of importing `astropy.tests.plugins.display` with a try/except is that is triggers a warning (`astropy.tests.plugins.display plugin has been deprecated.`).

Maybe we should just remove all the old stuff (astropy<4.0, astropy-helpers) ?

### CODE REVIEW

**Comment by pllim** [comment]
> Maybe we should just remove all the old stuff (astropy<4.0, astropy-helpers) ?

👍 to getting rid of old stuff. LTS is now 4.0.x. No reason why older stuff cannot pin `pytest-astropy-header`.

**Review by pllim — APPROVED** [review]
LGTM, but @astrofrog should have a look too.

We should pin `pytest-astropy-header` in `astropy` 4.0.x branch but I don't think that is a big deal. See astropy/astropy#11058

**Review by astrofrog — DISMISSED** [review]
Looks good to me but probably worthy of a changelog entry, and the astropy test dependency in setup.cfg should now have a minimum version.

File: CHANGES.rst
Comment (saimn) [inline CHANGES.rst:7]: Should I remove the other changelog entry (now obsolete) ? Or maybe we should do a release with this PR, and then another with the PR and bumping the version number to 0.2 ?

File: setup.cfg
Comment (saimn) [inline setup.cfg:31]: astropy is an optional requirement but the plugin does nothing if astropy is not installed. Is it to avoid a cross dependency which would confuse pip ?

Comment (pllim) [reply setup.cfg:31]: I noticed that too! I just thought since `astropy` is in the plugin name, it is not unreasonable to require `astropy` to be installed, but it did still catch me by surprise.

File: CHANGES.rst
Comment (astrofrog) [reply CHANGES.rst:7]: I agree with the second option

File: setup.cfg
Comment (pllim) [reply setup.cfg:31]: Let's address this part in a separate PR or issue.

**Review by pllim — APPROVED** [review]
Thanks!

File: CHANGES.rst
Comment (saimn) [reply CHANGES.rst:7]: @pllim - See above, we were thinking about doing a release without this PR, but I'm not sure who can do releases ;)
Maybe a good idea as well to setup an Action workflow to release on PyPI.

Comment (pllim) [reply CHANGES.rst:7]: Oh, ops. Should I revert?

Comment (pllim) [reply CHANGES.rst:7]: Looks like only @astrofrog can do the release.

Comment (saimn) [reply CHANGES.rst:7]: Maybe a good occasion to discuss releasing with an Action workflow for infrastructure packages ? ;)
https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/

Comment (astrofrog) [reply CHANGES.rst:7]: yes that would be great!

Comment (pllim) [reply CHANGES.rst:7]: I thought @astrofrog and @cadair already set up Azure for that? As much as I like Actions over Azure, I think the project should be consistent across the board.

Comment (saimn) [reply CHANGES.rst:7]: Well the Azure solution is a bit different, it uses cibuildwheel which is needed for wheels with compiled code, but for pure Python packages it is not.  Also the pypa/gh-action-pypi-publish action is to publish directly files (sdist and wheel) to PyPI, which would simplify the release process for pytest plugins etc.

Comment (pllim) [reply CHANGES.rst:7]: Ah, okay. Thanks for the clarifications!

Comment (Cadair) [reply CHANGES.rst:7]: It only uses cibuildwheel if you ask it to, it supports universal wheels and sdist without using it. In general tho I suggest we just use actions.

**Comment by pllim** [comment]
Following up on https://github.com/astropy/pytest-astropy-header/pull/32#discussion_r540629651 , for this particular package, it needs @astrofrog to work with @saimn to make it happen. FYI.

File: CHANGES.rst
Comment (saimn) [reply CHANGES.rst:7]: I was looking at doing a release before this change, but actually #16 was merged one year ago, and the removal of astropy-helpers code started before this PR (e.g. #29, #30). So maybe we can skip this release and go straight to 0.2.0 ?

Comment (pllim) [reply CHANGES.rst:7]: I am okay with whatever is easier to do.

**Pull request merged**
